src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -529,4 +534,28 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
             authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
         }
     }
+
+    // TODO: Add doc
+    private void setAuthorRoleOfPostings(List<Post> postsInCourse) {
+        // sets author roles to display user authority icon on posting headers
+        postsInCourse.forEach(post -> {
+            setAuthorRoleForPosting(post, post.getCoursePostBelongsTo());
+            post.getAnswers().forEach(answerPost -> setAuthorRoleForPosting(answerPost, post.getCoursePostBelongsTo()));
+        });
+    }
+
+    // TODO: Add doc
+    private void setAuthorRoleForPosting(Posting posting, Course postingCourse) {
+        User author = userRepository.findOneWithGroupsAndAuthoritiesById(posting.getAuthor().getId()).get();
+
+        if (authorizationCheckService.isAtLeastInstructorInCourse(postingCourse, author)) {
+            posting.setAuthorRole(UserRole.INSTRUCTOR);
+        }
+        else if (authorizationCheckService.isTeachingAssistantInCourse(postingCourse, author) || authorizationCheckService.isEditorInCourse(postingCourse, author)) {
```

Comment:
This should cover both.
```suggestion
        else if (authorizationCheckService.isAtLeastTeachingAssistantInCourse(postingCourse, author)) {
```

Reply:
I am aware of it but did not want to use this method, since it has unreachable code. isInstructor and isAdmin are checked and executed in the previous if statement.
![image](https://user-images.githubusercontent.com/67344158/181198718-3976fc8d-d6cd-4191-9473-b466859c5ae3.png)

Reply:
The code isn't unreachable (if the user is a student), but it would get executed twice. Considering that each "isXYInCourse()" call does a database query, I agree with the current implementation.